### ISSUE TITLE
API Definitions - Updating the API endpoint hostnames requires replacing the whole API Definition

### ISSUE THREAD

**Issue by vkolorizos** [akamai/AkamaiOPEN-edgegrid-golang#213]
Hello,

Currently, on the API Definition OpenAPI schema, when trying to update the hostnames on a new version of an API Endpoint the method required to provide all details and not just the required values, which is correct according to REST principles since it's a PUT request, but these raises some concerns since we will need to provide also the schema and multiple other fields, which might lead to unintended changes.

https://techdocs.akamai.com/api-definitions/reference/put-endpoint-version

In a nutshell, I feel this is neither efficient nor safe way to go, so maybe a specific-purpose PATCH request should be created to just update what part of the API endpoint someone wants or specific PATCH request for adding/removing hostnames like it exists already in Properties:

https://techdocs.akamai.com/property-mgr/reference/patch-property-version-hostnames

Let me know, what you think

Thanks

**Comment by vkolorizos** [comment]
Hello @aka-mark  and happy new year,

Any update about it? Is it something you are going to work on in Q1 or Q2 of 2025?

**Comment by lkowalsk-akamai-com** [comment]
@vkolorizos  - API Definitions endpoints are now available in this project.
